### PR TITLE
Change Arctic to Mixed for class names in module core

### PR DESCRIPTION
**Pull request by zhoujinsong**
## Why are the changes needed?

Arctic is no longer the name of the project.
In some old classes, Arctic is used to specify the mixed format, and this PR corrects the class names in the core module.

## Brief change log

- Change Arctic to Mixed for class names for module core.
- Change some comments.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by Aireed** [comment]
Some configuration items are named with "arctic." at the beginning.  should we change them too?

**Comment by zhoujinsong** [comment]
This PR does not change the thrift-related classes and table property keys, so it still maintains compatibility with older versions.

**Comment by zhoujinsong** [comment]
> @zhoujinsong Could you resolve the conflict if you had time?

I have merged master branch and fixed the conflicts, please task a look.

**Review by czy006 — COMMENTED** [review]
I think some of the key-value configurations should also be amoro and internally compatible. In the future v1.0 we will completely remove this compatibility.WHDY? @zhoujinsong

**Comment by zhoujinsong** [comment]
> I think some of the key-value configurations should also be amoro and internally compatible. In the future v1.0 we will completely remove this compatibility.WHDY? @zhoujinsong

Yes, we may push forward this in another PR after discussion in the dev mail list.

File: amoro-core/src/main/thrift/amoro_commons.thrift
Comment (zhongqishang) [inline amoro-core/src/main/thrift/amoro_commons.thrift:1]: The Thrift file name has been changed, and I noticed that there are still some "Arctic" characters in the code, and compiler failed.

Comment (zhongqishang) [reply amoro-core/src/main/thrift/amoro_commons.thrift:1]: I recommend rolling back the Thrift filename and as a separate PR to fix it. WDYT?

**Comment by zhongqishang** [comment]
All class name were changed from `Arctic` to `Mixed` in the core module?  IMO, `MixedFileIO` looks weird, what does mixed represent?

File: amoro-core/src/main/thrift/amoro_commons.thrift
Comment (zhoujinsong) [reply amoro-core/src/main/thrift/amoro_commons.thrift:1]: I have rollback the change about thrift files.

**Comment by zhoujinsong** [comment]
> All class name were changed from `Arctic` to `Mixed` in the core module? IMO, `MixedFileIO` looks weird, what does mixed represent?

`ArcticFileIO` was initially used to implement mixed format tables, but as it evolved, it is now also employed to provide authenticated FileIO implementations for the Iceberg format. So how about renaming `ArcticFileIO` to `AuthenticatedFileIO`？

**Review by zhongqishang — APPROVED** [review]
LGTM. Thanks for your contribution. 
Approve first and merge after resolving conflicts.

**Pull request merged**